### PR TITLE
Fix CLERK_RETHEME build issue on clerk-js

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description

Introduce the `~ui` path alias on the build and type level (webpack and tsconfig) so we can conditionally build with or without the new UI components. The new alias was needed because we `paths` in tsconfig will not work with relative imports, so we could not alias the top-level imports on `index.ts` and `index.browser.ts`

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/tsconfig.json [hunk 1]
```diff
@@ -20,8 +20,10 @@
     "useUnknownInCatchVariables": false,
     "declaration": false,
     "jsx": "react-jsx",
-    "jsxImportSource": "@emotion/react"
-    // "noUncheckedIndexedAccess": true // TODO: Enable and progressively fix related errors
```
Comment (panteliselef) [inline packages/clerk-js/tsconfig.json:24]: @nikosdouvlis Shall we add a ticket to bring `noUncheckedIndexedAccess` back ?

File: packages/clerk-js/src/ui.retheme/components/OrganizationProfile/BillingWidget.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { useRouter } from '../../../ui/router';
 import { runIfFunctionOrReturn } from '../../../utils';
 import { AlertIcon, Flex, Link, Text } from '../../customizables';
+import { useRouter } from '../../router';
```
Comment (panteliselef) [inline packages/clerk-js/src/ui.retheme/components/OrganizationProfile/BillingWidget.tsx:3]: Now that we have the alias isn't it better to do `from '~ui/router'` ?

**Review by dimkl — COMMENTED** [review]
❓ Instead of creating the alias, could we move those to `utils`?

File: packages/clerk-js/tsconfig.json (hunk 1 above)
Comment (nikosdouvlis) [reply packages/clerk-js/tsconfig.json:24]: This wasn't changed as part of this PR but agreed, let's add a ticket in triage

**Review by desiprisg — APPROVED** [review]

**Comment by nikosdouvlis** [comment]
> ❓ Instead of creating the alias, could we move those to `utils`?

No, as even if we moved them, we'd still need to conditionally import from `ui` or `ui.retheme` at some point, so it'd be the same.

**Pull request merged**